### ISSUE TITLE
Sync function panics when it encounters "%~d" in an object name.

### ISSUE THREAD

**Issue by cdiskan** [Azure/azure-storage-azcopy#2175]
##### Note: The version is visible when running AzCopy without any argument
10.18.0

### Which platform are you using? (ex: Windows, Mac, Linux)
Windows

##### Note: Please remove the SAS to avoid exposing your credentials. If you cannot remember the exact command, please retrieve it from the beginning of the log file.

.\azcopy.exe sync $source $DestinationSASstring --delete-destination=True --mirror-mode --preserve-permissions=true

### What problem was encountered?
azcopy.exe sync panics when it encounters a file or folder with %~d in the name. 

Error provided in console:
.\azcopy.exe : panic: invalid URL escape "%~d"
At C:\Users\USERNAME\Desktop\Sync via Azcopy - v1 - DDOO.ps1:22 char:1
+ .\azcopy.exe sync $source $DestinationSASstring --delete-destination= ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (panic: invalid URL escape "%~d":String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
 
goroutine 1 [running]:
github.com/Azure/azure-storage-azcopy/v10/common.PanicIfErr
(...)
	/home/vsts/work/1/s/common/lifecyleMgr.go:696
github.com/Azure/azure-storage-azcopy/v10/cmd.(*copyTransferProcessor).scheduleCopyTransfer.func1
(0xf09cc408?)
	/home/vsts/work/1/s/cmd/zc_processor.go
:
108 +0x8a9
github.com/Azure/azure-storage-azcopy/v10/common.(*lifecycleMgr).Dryrun(0xc0002da900
, 0x35?
)
	/home/vsts/work/1/s/common/lifecyleMgr.go:327 +0x31
github.com/Azure/azure-storage-azcopy/v10/cmd.(*copyTransferProcessor).scheduleCopyTransfer
(
_
, {{0xc0015ed600, 0x18}
, 
0x0, {0x313da7a8
, 0xed94acfa4
, 0x2899120}
, {
0x0, 0x0
, 0x0}, ...})
	/home/vsts/work/1/s/cmd/zc_processor.go:101 +0x5df
github.com/Azure/azure-storage-azcopy/v10/cmd.(*syncDestinationComparator).processIfNecessary
(_, {{0xc16991f080, 
0x18}, 0x0, {0x0, 0xedb8f0722, 0xc1ec4c1f10}, {0x313da7a8, 0xed94acfa4, 0x0}, ...})
	/home/vsts/work/1/s/cmd/syncComparator.go:93 +
0x562
github.com/Azure/azure-storage-azcopy/v10/cmd.processIfPassedFilters(
{
_, _, _
}, 
{
{0xc16991f080, 0x18}, 0x0, 
{0x0, 
0xedb8f0722, 0xc1ec4c1f10
}, 
{
0x313da7a8, ...}
, ...}, ...
)
	/home/vsts/work/1/s/cmd/zc_enumerator.go:771 +0x9f
github.com/Azure/azure-storage-azcopy/v10/cmd.(*fileTraverser).Traverse.func2
(
{{0xc16991f080, 
0x18}, 0x0, {0x0, 0xedb8f0722, 0xc1ec4c1f10}, {0x313da7a8, 0xed94acfa4, 0x0}, 0x17b124b, ...})
	/home/vsts/work/1/s/cmd/zc_traverser_file.go:165 +0xbb
github.com/Azure/azure-storage-azcopy/v10/cmd.(*fileTraverser).Traverse(0xc000152240, 0x0, 0xc0000b03e0, 
{0x28f13c8, 0x0
, 0x0})
	/home/vsts/work/1/s/cmd/zc_traverser_file.go:258 +0xd23
github.com/Azure/azure-storage-azcopy/v10/cmd.(*syncEnumerator).enumerate(
0xc0005b0230
)
	/home/vsts/work/1/s/cmd/zc_enumerator.go:
674
 +
0xb0
github.com/Azure/azure-storage-azcopy/v10/cmd.(*cookedSyncCmdArgs).process(0xc000186000
)
	
/home/vsts/work/1/s/cmd/sync.go
:
726 +
0x672
github.com/Azure/azure-storage-azcopy/v10/cmd.init.20.func2(0xc000763900?, {
0xc0001916e0?
, 0x2?, 0x6
?})
	
/home/vsts/work/1/s/cmd/sync.go
:
762 +0x211
github.com/spf13/cobra.(*Command).execute
(0xc000763900, {0xc000191620, 0x6, 0x6})
	/home/vsts/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:860 +0x663
github.com/spf13/cobra.(*Command).ExecuteC(0x288a1a0)
	/home/vsts/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:974 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
	/home/vsts/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:902
github.com/Azure/azure-storage-azcopy/v10/cmd.Execute({
0xc0000349e0?
, 
0xc0001080e0?}, {0xc00002e450?, 0x200fa4d
?
}
, 0x74?, {0xa618561c, 0xd2f8
, 0x614a
, {0x50
, 
0x6c, ...}})
	/home/vsts/work/1/s/cmd/root.go
:192 +0x106
main.main(
)
	/home/vsts/work/1/s/main.go:87
 +
0x4eb

### How can we reproduce the problem in the simplest way?
I suspect you can replicate this by creating a file in Azure Files with a name that includes "%~d" and then trying to remove it via an azcopy sync job that would overwrite it, but I have not attempted to replicate it in a lab yet. 

### Have you found a mitigation/solution?
Manually delete or rename the problem file or directory. (This can be a challenge, as it doesn't seem like the problem file gets logged when AZCopy panics).

**Comment by gapra-msft** [comment]
Hi @cdiskan Could you please share a full example path of a file so we can reproduce this issue? Is the file in question in Azure File or local?

**Comment by gapra-msft** [comment]
Closing due to inactivity in the thread. Feel free to reopen the issue once you have an answer for our team in case you are still experiencing this issue.

**Comment by cdiskan** [comment]
Apologies. We ended up deleting these files as they were holding up production migrations. We identified the files with the following PowerShell, and then removed them. 

`Get-ChildItem  -Recurse -File -Filter "*%~d*"`

The file were on a local windows file server, being copied to an AZ Files share. 

It is unknown where they came from, and some had creation dates as far back as the early 1990s. I suspect that based on the filenames some of them were echoed outputs of batch files as they included both and  "%~dpo" (Used in batch to get the current directory) and "%~dp0" (A logical typo). 

In a quick test, I was unable to create a new file with the invalid name on modern systems. Hopefully this closed issue is enough for anyone else who runs into the issue to sort it out.
